### ISSUE TITLE
npm run start goes into error

### ISSUE THREAD

**Issue by karlcow** [mozilla/webcompat-addon#139]
Steps to reproduce.

```
git clone git://github.com/mozilla/webcompat-addon
npm install
# modify ua-override.js
npm run prettier
# everything is good.
npm run start

> webcompat-gofaster@1.0.0 start /Users/karl/code/webcompat-addon
> npx web-ext run --source-dir src/

Applying config file: ./package.json
Running web extension from /Users/karl/code/webcompat-addon/src
Use --verbose or open Tools > Web Developer > Browser Console to see logging

o: installTemporaryAddon: Error: unknownError: Could not install add-on at '/Users/karl/code/webcompat-addon/src': Error: Add-on webcompat@mozilla.org is not compatible with application version. add-on minVersion: 59.0b5.
    at /Users/karl/code/webcompat-addon/node_modules/web-ext/dist/web-ext.js:1:5967
    at Client.handleMessage (/Users/karl/code/webcompat-addon/node_modules/@cliqz-oss/firefox-client/lib/client.js:162:7)
    at Client.readMessage (/Users/karl/code/webcompat-addon/node_modules/@cliqz-oss/firefox-client/lib/client.js:221:10)
    at Client.onData (/Users/karl/code/webcompat-addon/node_modules/@cliqz-oss/firefox-client/lib/client.js:187:16)
    at Socket.emit (events.js:209:13)
    at addChunk (_stream_readable.js:305:12)
    at readableAddChunk (_stream_readable.js:286:11)
    at Socket.Readable.push (_stream_readable.js:220:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:182:23)

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! webcompat-gofaster@1.0.0 start: `npx web-ext run --source-dir src/`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the webcompat-gofaster@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/karl/.npm/_logs/2020-06-05T02_44_07_643Z-debug.log
```

some more information.

```
nvm ls                                             
->     v12.10.0
         system
default -> v12.10.0
node -> stable (-> v12.10.0) (default)
stable -> 12.10 (-> v12.10.0) (default)
iojs -> N/A (default)
unstable -> N/A (default)
nvm_list_aliases:36: no matches found: /Users/karl/.nvm/alias/lts/*
```

```
git log --oneline
c32854d (HEAD -> master, origin/master, origin/HEAD) Merge pull request #137 from mozilla/1631858/3
b6e8d9e Bug 1546051 - Hide PiP on Netflix previews
76d5798 Merge pull request #136 from mozilla/1631858/2
4a54d5f Bug 1633796 - Add CSS intervention for www.laposte.fr login
…
```

**Comment by karlcow** [comment]
```
more /Users/karl/.npm/_logs/2020-06-05T02_44_07_643Z-debug.log
0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   '/Users/karl/.nvm/versions/node/v12.10.0/bin/node',
1 verbose cli   '/Users/karl/.nvm/versions/node/v12.10.0/bin/npm',
1 verbose cli   'run',
1 verbose cli   'start'
1 verbose cli ]
2 info using npm@6.14.3
3 info using node@v12.10.0
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle webcompat-gofaster@1.0.0~prestart: webcompat-gofaster@1.0.0
6 info lifecycle webcompat-gofaster@1.0.0~start: webcompat-gofaster@1.0.0
7 verbose lifecycle webcompat-gofaster@1.0.0~start: unsafe-perm in lifecycle true
8 verbose lifecycle webcompat-gofaster@1.0.0~start: PATH: /Users/karl/.nvm/versions/node/v12.10.0/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/Users/karl/code/webcompat-addon/node_modules/.bin:/Users/karl/.nvm/versions/node/v12.10.0/bin:/Applications/Firefox.app/Contents/MacOS:/Users/karl/bin:/usr/local/bin: etc…
9 verbose lifecycle webcompat-gofaster@1.0.0~start: CWD: /Users/karl/code/webcompat-addon
10 silly lifecycle webcompat-gofaster@1.0.0~start: Args: [ '-c', 'npx web-ext run --source-dir src/' ]
11 silly lifecycle webcompat-gofaster@1.0.0~start: Returned: code: 1  signal: null
12 info lifecycle webcompat-gofaster@1.0.0~start: Failed to exec start script
13 verbose stack Error: webcompat-gofaster@1.0.0 start: `npx web-ext run --source-dir src/`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/Users/karl/.nvm/versions/node/v12.10.0/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:209:13)
13 verbose stack     at ChildProcess.<anonymous> (/Users/karl/.nvm/versions/node/v12.10.0/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:209:13)
13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
14 verbose pkgid webcompat-gofaster@1.0.0
15 verbose cwd /Users/karl/code/webcompat-addon
16 verbose Darwin 19.5.0
17 verbose argv "/Users/karl/.nvm/versions/node/v12.10.0/bin/node" "/Users/karl/.nvm/versions/node/v12.10.0/bin/npm" "run" "start"
18 verbose node v12.10.0
19 verbose npm  v6.14.3
20 error code ELIFECYCLE
21 error errno 1
22 error webcompat-gofaster@1.0.0 start: `npx web-ext run --source-dir src/`
22 error Exit status 1
23 error Failed at the webcompat-gofaster@1.0.0 start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

**Comment by denschub** [comment]
> Error: Add-on webcompat@mozilla.org is not compatible with application version. add-on minVersion: 59.0b5.

What browser/device are you trying to install this to?

`web-ext` isn't really meant as the default testing, which is why it's not mentioned in the `README.md` except for testing on Fennec, but even there, this isn't meant to be used as "just run the command", because `web-ext` may require additional params.

On desktop, you should be fine just following https://github.com/mozilla/webcompat-addon#via-aboutdebugging

**Comment by miketaylr** [comment]
> `web-ext` isn't really meant as the default testing, which is why it's not mentioned in the `README.md` except for testing on Fennec, but even there, this isn't meant to be used as "just run the command", because `web-ext` may require additional params.

That's on me, I was trying to show Karl how to do an example UA override last night and recommended web-ext (because when it does work, it's much nicer UX than about:debugging).

**Comment by denschub** [comment]
Since no details were provided, I *assume* this was the issue fixed in web-ext 4.2.0: https://github.com/mozilla/web-ext/releases/tag/4.2.0 I added a update to the v12 PR, and once that is merged, this issue will auto-close. If this doesn't resolve, please reopen and elaborate on which Firefox-version you're testing on.

For the record, it's still unlikely you can run `npm run start`, because that might start an older version if you have multiple Firefoxes on your machine. Use something like `npm run start -- --firefox=nightly` to run in Firefox Desktop Nightly. Check the [`web-ext` docs](https://extensionworkshop.com/documentation/develop/web-ext-command-reference/) for a full reference on available flags, and ping me in Slack or Matrix if you have issues.
